spec/services/green_lanes/fetch_goods_nomenclature_spec.rb

```diff
@@ -19,21 +19,27 @@
   end
 
   describe '#call' do
-    it 'fetches the goods nomenclature with the correct parameters', :aggregate_failures do
-      service = described_class.new(params)
-      result = service.call
+    subject(:fetch_goods) { described_class.new(params).call }
 
-      expect(GreenLanes::GoodsNomenclature).to have_received(:find).with(
-        params[:commodity_code],
-        {
-          filter: {
-            geographical_area_id: params[:country_of_origin],
-            moving_date: params[:moving_date],
+    context 'with valid parameters' do
+      it 'calls GoodsNomenclature.find with the correct arguments' do
+        fetch_goods
+        expect(GreenLanes::GoodsNomenclature).to have_received(:find).with(
+          params[:commodity_code],
+          {
+            filter: {
+              geographical_area_id: params[:country_of_origin],
+              moving_date: params[:moving_date],
+            },
+            as_of: params[:moving_date],
           },
-          authorization: 'test_token',
-        },
-      )
-      expect(result).to eq(expected_response)
+          authorization: TradeTariffFrontend.green_lanes_api_token,
+        )
+      end
+
+      it 'returns the expected response' do
```

Comment:
I am aware that you did not start this pattern, 
return the `returns the expected response` is not such helpful description :smile_cat: